extensions/PhabBugz/lib/Util.pm

```diff
@@ -265,11 +283,11 @@ sub request {
     ThrowCodeError('phabricator_api_error', { reason => $response->message })
         if $response->is_error;
 
-    my $result = decode_json($response->content);
-    if ($result->{error_code}) {
-        ThrowCodeError('phabricator_api_error',
-                       { code   => $result->{error_code},
-                         reason => $result->{error_info} });
+    my $result = eval { decode_json( $response->content ) };
+    if ( !$result || $@ ) {
```

Comment:
Pretty sure you just want either $@ (although in rare pathological cases it can be cleared, because of DESTROY handlers) or defined $result.
To be super paranoid about someone passing 'null' as a json value...
```perl
my $result;
my $result_ok = eval { $result = decode_json( $response->content ); 1 };
if ( !$result_ok) {
```